x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/multi_field_selector.tsx

```diff
@@ -0,0 +1,221 @@
+
+import {
+  EuiBasicTable,
+  EuiBasicTableColumn,
+  EuiButton,
+  EuiComboBox,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiText,
+} from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { FieldMapping } from '../../../../../../../common/ml_inference_pipeline';
+
+import { IndexViewLogic } from '../../index_view_logic';
+
+import { MLInferenceLogic } from './ml_inference_logic';
+
+type FieldNames = Array<{ label: string }>;
+
+export const MultiFieldMapping: React.FC = () => {
+  const {
+    addInferencePipelineModal: { configuration, selectedSourceFields = [] },
+    sourceFields,
+  } = useValues(MLInferenceLogic);
+  const { ingestionMethod } = useValues(IndexViewLogic);
+  const { addSelectedFieldsToMapping, selectFields } = useActions(MLInferenceLogic);
+
+  const mappedSourceFields =
+    configuration.fieldMappings?.map(({ sourceField }) => sourceField) ?? [];
+
+  // Remove fields that have already been selected or mapped from selectable field options
+  const fieldOptions = (sourceFields || [])
+    .filter((fieldName) => ![...selectedSourceFields, ...mappedSourceFields].includes(fieldName))
+    .map((fieldName) => ({ label: fieldName }));
+
+  const selectedFields = selectedSourceFields.map((fieldName) => ({
+    label: fieldName,
+  }));
+
+  const onChangeSelectedFields = (selectedFieldNames: FieldNames) => {
+    selectFields(selectedFieldNames.map(({ label }) => label));
+  };
+
+  const onCreateField = (fieldName: string) => {
+    const normalizedFieldName = fieldName.trim();
+    if (!normalizedFieldName) return;
+
+    selectedFields.push({ label: normalizedFieldName });
+    selectFields([...selectedSourceFields, fieldName]);
+  };
+
+  return (
+    <>
+      <EuiFlexGroup>
+        <EuiFlexItem grow={4}>
+          <EuiFormRow
+            fullWidth
+            label={i18n.translate(
+              'xpack.enterpriseSearch.content.indices.pipelines.addInferencePipelineModal.steps.fields.sourceFieldLabel',
+              {
+                defaultMessage: 'Source field',
+              }
+            )}
+            helpText="Select an existing field or type in a field name."
+          >
+            <EuiComboBox
+              fullWidth
+              data-telemetry-id={`entSearchContent-${ingestionMethod}-pipelines-configureFields-selectSchemaField`}
+              placeholder={i18n.translate(
+                'xpack.enterpriseSearch.content.indices.pipelines.addInferencePipelineModal.steps.fields.selectedFields',
+                {
+                  defaultMessage: 'Selected fields',
+                }
+              )}
+              options={fieldOptions}
+              selectedOptions={selectedFields}
+              onChange={onChangeSelectedFields}
+              onCreateOption={onCreateField}
+            />
+          </EuiFormRow>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiText style={{ paddingTop: '25px' }}>
+            <p>-&gt;</p>
```

Comment:
Should this be an `<EuiIcon>` with `arrowRight` ? @julianrosado

Reply:
`sortRight` !

![CleanShot 2023-04-11 at 13 09 11](https://user-images.githubusercontent.com/4490333/231238108-6c45af5e-f96e-45df-b59c-637c55ae911e.png)